sstables/sstables.cc

```diff
@@ -2157,41 +2157,42 @@ future<> sstable::filesystem_storage::check_create_links_replay(const sstable& s
 /// are used.  These versions handle EEXIST errors that may happen
 /// when the respective operations are replayed.
 ///
+/// \param sst - the sstable to work on
 /// \param dst_dir - the destination directory.
 /// \param generation - the generation of the destination sstable
 /// \param mark_for_removal - mark the sstable for removal after linking it to the destination dst_dir
-future<> sstable::create_links_common(sstring dst_dir, generation_type generation, mark_for_removal mark_for_removal) const {
-    sstlog.trace("create_links: {} -> {} generation={} mark_for_removal={}", get_filename(), dst_dir, generation, mark_for_removal);
-    auto comps = all_components();
-    co_await _storage.check_create_links_replay(*this, dst_dir, generation, comps);
+future<> sstable::filesystem_storage::create_links_common(const sstable& sst, sstring dst_dir, generation_type generation, mark_for_removal mark_for_removal) const {
+    sstlog.trace("create_links: {} -> {} generation={} mark_for_removal={}", sst.get_filename(), dst_dir, generation, mark_for_removal);
+    auto comps = sst.all_components();
+    co_await check_create_links_replay(sst, dst_dir, generation, comps);
     // TemporaryTOC is always first, TOC is always last
-    auto dst = sstable::filename(dst_dir, _schema->ks_name(), _schema->cf_name(), _version, generation, _format, component_type::TemporaryTOC);
-    co_await sstable_write_io_check(idempotent_link_file, filename(component_type::TOC), std::move(dst));
-    co_await sstable_write_io_check(sync_directory, dst_dir);
-    co_await parallel_for_each(comps, [this, &dst_dir, generation] (auto p) {
-        auto src = sstable::filename(_storage.dir, _schema->ks_name(), _schema->cf_name(), _version, _generation, _format, p.second);
-        auto dst = sstable::filename(dst_dir, _schema->ks_name(), _schema->cf_name(), _version, generation, _format, p.second);
-        return sstable_write_io_check(idempotent_link_file, std::move(src), std::move(dst));
+    auto dst = sstable::filename(dst_dir, sst._schema->ks_name(), sst._schema->cf_name(), sst._version, generation, sst._format, component_type::TemporaryTOC);
+    co_await sst.sstable_write_io_check(idempotent_link_file, sst.filename(component_type::TOC), std::move(dst));
```

Comment:
shouldn't `sstable_write_io_check` move to storage as well?

Reply:
Yes, eventually. If I do it now, it will be copied between sstable and filsystem_storage. Maybe it's not that bad, but I wanted to check if it could completely live in the latter one.